tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml

```diff
@@ -150,12 +150,27 @@
 
     ## Cleanup and fail gracefully
     rescue:
-    - name: Include rescue from terraform failure
-      ansible.builtin.include_tasks: "tasks/rescue_terraform_failure.yml"
+    - name: Gather logs
+      ansible.builtin.include_tasks:
+        file: tasks/gather_logs.yml
+        apply:
+          delegate_to: localhost
+      vars:
+        project: "{{ project }}"
+        remote_node: "{{ remote_node }}"
```

Comment:
Using the remote_node is not sufficient in all cases. There is no guarantee that wait-for-startup is waiting on the remote_node. The original implementation pulls the gcloud command from the output reason, so that we pull logs from the machine where the failure occured. What machine we are waiting on is embedded in the blueprint and not really knowable until terraform apply.

Example: Batch mpi where remote node is the login node but wait-for-startup waits on builder.

Reply:
Point taken. Will document this as a task. I believe the right solution is to ensure that the relevant terraform modules have an output which includes the resource name and allow us to construct instructions like the gcloud command directly instead of via regex searches.

Re-worked PR to address only guarantee that terraform destroy is reached without addressing when Ansible crashes and burns if jinja can't resolve the command to invoke gcloud for startup scripts.